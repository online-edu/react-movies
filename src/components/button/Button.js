import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button component.
 */
const Button = ({ aria, tabIndex, label, click }) => (
    <button
        type="button"
        onClick={click}
        aria-label={label}
        tabIndex={tabIndex}
        {...aria}
        className="btn btn-primary my-2"
    >
        {label}
    </button>
);
/**
 * Button props types.
 */
Button.propTypes = {
    /** Id to be assigned to the control. */
    aria: PropTypes.object,
    /** Display value */
    label: PropTypes.string.isRequired,
    /** Gets called when the user clicks on button */
    click: PropTypes.func.isRequired,
    /** Maintain tab index for keyboard navigation. */
    tabIndex: PropTypes.number,
};
export default Button;
