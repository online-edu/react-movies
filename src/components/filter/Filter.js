import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Collapse from 'react-bootstrap/Collapse';

/**
 * Filter component.
 */
class Filter extends Component {
    static get propTypes() {
        return {
            /** Gets called when the user clicks on the 'Apply' button */
            onChange: PropTypes.func.isRequired,
        };
    }
    /**
     * Intilalizes the state and binds all methods.
     *
     * @param {Object} props - access 'props'
     */
    constructor(props) {
        super(props);
        this.state = {
            open: false,
            enLangFilter: false,
            adultFilter: false,
            ratingFilter: 0,
        };
        this.handleInputChange = this.handleInputChange.bind(this);
        this.onFilterChange = this.onFilterChange.bind(this);
    }
    /**
     * Prepares filter options and pass to the parent component.
     */
    onFilterChange() {
        const {
            ratingFilter: vote_average,
            enLangFilter: original_language,
            adultFilter: adult,
        } = this.state;
        const { onChange } = this.props;
        const commonFilters = { vote_average, adult };
        const filters = original_language
            ? { ...commonFilters, original_language: 'en' }
            : commonFilters;
        onChange(filters);
    }
    /**
     * Updates the state as user updates the filter options.
     *
     * @param {Event} e - an event from dom controls
     */
    handleInputChange(e) {
        const { target } = e;
        const { type, name } = target;
        const value = type === 'checkbox' ? target.checked : target.value;
        this.setState({ [name]: value });
    }
    /**
     * Render method for component
     */
    render() {
        const { open, ratingFilter } = this.state;
        return (
            <div className="bg-secondary px-3 rounded">
                <section className="d-flex justify-content-between align-items-center">
                    <h6 className="mb-0">Popular movies</h6>
                    <button
                        type="button"
                        onClick={() => this.setState({ open: !open })}
                        aria-controls="movie-filter"
                        aria-label="Filter"
                        aria-expanded={open}
                        tabIndex="2"
                        className="btn btn-primary my-2"
                    >
                        Filter
                    </button>
                </section>
                <Collapse in={open}>
                    <div
                        id="movie-filter"
                        role="toolbar"
                        aria-label="Filter options"
                        className="border-top border-primary mt-1 py-3 px-1"
                    >
                        <h6 className="text-uppercase">Filter options</h6>
                        <div className="d-flex flex-wrap justify-content-between align-items-center">
                            <div>
                                <label
                                    htmlFor="rating-filter"
                                    aria-label="Rating filter"
                                    aria-describedby="rating-filter"
                                >
                                    Rating (&ge;)
                                    <input
                                        type="range"
                                        className="custom-range"
                                        aria-label="Rating range"
                                        onChange={this.handleInputChange}
                                        name="ratingFilter"
                                        min="0"
                                        max="10"
                                        step="0.5"
                                        value={ratingFilter}
                                        id="rating-filter"
                                        tabIndex="3"
                                    />
                                    <span className="text-primary position-absolute font-weight-bold ml-2">
                                        {ratingFilter}
                                    </span>
                                </label>
                            </div>
                            <div className="custom-control custom-switch mt-4">
                                <input
                                    type="checkbox"
                                    className="custom-control-input"
                                    aria-label="Adult Checkbox"
                                    id="adult-filter"
                                    name="adultFilter"
                                    onChange={this.handleInputChange}
                                    tabIndex="4"
                                />
                                <label
                                    className="custom-control-label"
                                    htmlFor="adult-filter"
                                    aria-label="Adult filter"
                                    aria-describedby="adult-filter"
                                >
                                    Adult
                                </label>
                            </div>
                            <div className="custom-control custom-switch mt-4">
                                <input
                                    type="checkbox"
                                    className="custom-control-input"
                                    onChange={this.handleInputChange}
                                    aria-label="Original Language Checkbox"
                                    name="enLangFilter"
                                    id="en-lang-filter"
                                    tabIndex="5"
                                />
                                <label
                                    className="custom-control-label"
                                    htmlFor="en-lang-filter"
                                    aria-label="Original Language English filter"
                                    aria-describedby="en-lang-filter"
                                >
                                    Original Language English
                                </label>
                            </div>
                        </div>
                        <div className="mt-2">
                            <button
                                type="button"
                                aria-label="Apply"
                                onClick={this.onFilterChange}
                                className="btn btn-primary"
                                tabIndex="6"
                            >
                                Apply
                            </button>
                        </div>
                    </div>
                </Collapse>
            </div>
        );
    }
}

export default Filter;
