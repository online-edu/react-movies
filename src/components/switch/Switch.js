import React from 'react';
import PropTypes from 'prop-types';

/**
 * Switch component.
 */
const Switch = ({ switchToggle, id, tabIndex, label }) => (
    <div className="custom-control custom-switch">
        <input
            type="checkbox"
            className="custom-control-input"
            aria-label={`${label} Checkbox`}
            id={id}
            name={id}
            onChange={switchToggle}
            tabIndex={tabIndex}
        />
        <label
            className="custom-control-label"
            htmlFor={id}
            aria-label={`${label} filter`}
            aria-describedby={id}
        >
            {label}
        </label>
    </div>
);
/**
 * Switch props types.
 */
Switch.propTypes = {
    /** Id to be assigned to the control. */
    id: PropTypes.number.isRequired,
    /** Display value */
    label: PropTypes.string.isRequired,
    /** Gets called when the user clicks on switch */
    switchToggle: PropTypes.func.isRequired,
    /** Maintain tab index for keyboard navigation. */
    tabIndex: PropTypes.number.isRequired,
};
export default Switch;
