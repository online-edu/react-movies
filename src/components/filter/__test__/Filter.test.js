import React from 'react';
import { shallow } from 'enzyme';
import Filter from '../Filter';
import Button from '../../button';
import Switch from '../../switch';

describe('<Filter />', () => {
    it('renders correctly', () => {
        const component = shallow(
            <Filter onChange={jest.fn()} />,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders two <Button /> components', () => {
        const wrapper = shallow(
            <Filter onChange={jest.fn()} />,
        );
        expect(wrapper.find(Button)).toHaveLength(2);
    });

    it('renders two <Switch /> components', () => {
        const wrapper = shallow(
            <Filter onChange={jest.fn()} />,
        );
        expect(wrapper.find(Switch)).toHaveLength(2);
    });

    it('expects "event" from dom objeect', () => {
        const wrapper = shallow(<Filter onChange={jest.fn()} />);
        const e = {
            target: {
                checked: true,
                type: 'checkbox',
                name: 'adultFilter',
            },
        };
        wrapper.instance().handleInputChange(e);
        expect(wrapper.state().adultFilter).toBe(true);
    });

    it('triggers "filterChange" on "apply" click', () => {
        const onFilterChange = jest.fn();
        const wrapper = shallow(<Filter onChange={onFilterChange} />);
        wrapper.instance().onFilterChange();
        wrapper.simulate('onChange');
        expect(onFilterChange).toHaveBeenCalled();
        expect(onFilterChange).toHaveBeenCalledTimes(1);
        expect(onFilterChange).toHaveBeenCalledWith({
            adult: false,
            vote_average: 0,
        });
    });
});
