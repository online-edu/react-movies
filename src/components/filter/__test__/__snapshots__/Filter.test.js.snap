// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="d-flex justify-content-between align-items-center"
        >
          <h6
            className="mb-0"
          >
            Popular movies
          </h6>
          <Button
            aria={
              Object {
                "aria-controls": "movie-filter",
                "aria-expanded": false,
              }
            }
            click={[Function]}
            label="Filter"
          />
        </section>,
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            aria-label="Filter options"
            className="border-top border-primary mt-1 py-3 px-1"
            id="movie-filter"
            role="toolbar"
          >
            <h6
              className="text-uppercase"
            >
              Filter options
            </h6>
            <div
              className="d-flex flex-wrap justify-content-between align-items-center"
            >
              <div>
                <label
                  aria-describedby="rating-filter"
                  aria-label="Rating filter"
                  htmlFor="rating-filter"
                >
                  Rating (≥)
                  <input
                    aria-label="Rating range"
                    className="custom-range"
                    id="rating-filter"
                    max="10"
                    min="0"
                    name="ratingFilter"
                    onChange={[Function]}
                    step="0.5"
                    type="range"
                    value={0}
                  />
                  <span
                    className="text-primary position-absolute font-weight-bold ml-2"
                  >
                    0
                  </span>
                </label>
              </div>
              <Switch
                id="adultFilter"
                label="Adult"
                switchToggle={[Function]}
              />
              <Switch
                id="enLangFilter"
                label="Original Language"
                switchToggle={[Function]}
              />
            </div>
            <div
              className="mt-1"
            >
              <Button
                aria={Object {}}
                click={[Function]}
                label="Apply"
              />
            </div>
          </div>
        </Collapse>,
      ],
      "className": "bg-secondary px-3 rounded",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h6
              className="mb-0"
            >
              Popular movies
            </h6>,
            <Button
              aria={
                Object {
                  "aria-controls": "movie-filter",
                  "aria-expanded": false,
                }
              }
              click={[Function]}
              label="Filter"
            />,
          ],
          "className": "d-flex justify-content-between align-items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Popular movies",
              "className": "mb-0",
            },
            "ref": null,
            "rendered": "Popular movies",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria": Object {
                "aria-controls": "movie-filter",
                "aria-expanded": false,
              },
              "click": [Function],
              "label": "Filter",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <div
            aria-label="Filter options"
            className="border-top border-primary mt-1 py-3 px-1"
            id="movie-filter"
            role="toolbar"
          >
            <h6
              className="text-uppercase"
            >
              Filter options
            </h6>
            <div
              className="d-flex flex-wrap justify-content-between align-items-center"
            >
              <div>
                <label
                  aria-describedby="rating-filter"
                  aria-label="Rating filter"
                  htmlFor="rating-filter"
                >
                  Rating (≥)
                  <input
                    aria-label="Rating range"
                    className="custom-range"
                    id="rating-filter"
                    max="10"
                    min="0"
                    name="ratingFilter"
                    onChange={[Function]}
                    step="0.5"
                    type="range"
                    value={0}
                  />
                  <span
                    className="text-primary position-absolute font-weight-bold ml-2"
                  >
                    0
                  </span>
                </label>
              </div>
              <Switch
                id="adultFilter"
                label="Adult"
                switchToggle={[Function]}
              />
              <Switch
                id="enLangFilter"
                label="Original Language"
                switchToggle={[Function]}
              />
            </div>
            <div
              className="mt-1"
            >
              <Button
                aria={Object {}}
                click={[Function]}
                label="Apply"
              />
            </div>
          </div>,
          "dimension": "height",
          "getDimensionValue": [Function],
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Filter options",
            "children": Array [
              <h6
                className="text-uppercase"
              >
                Filter options
              </h6>,
              <div
                className="d-flex flex-wrap justify-content-between align-items-center"
              >
                <div>
                  <label
                    aria-describedby="rating-filter"
                    aria-label="Rating filter"
                    htmlFor="rating-filter"
                  >
                    Rating (≥)
                    <input
                      aria-label="Rating range"
                      className="custom-range"
                      id="rating-filter"
                      max="10"
                      min="0"
                      name="ratingFilter"
                      onChange={[Function]}
                      step="0.5"
                      type="range"
                      value={0}
                    />
                    <span
                      className="text-primary position-absolute font-weight-bold ml-2"
                    >
                      0
                    </span>
                  </label>
                </div>
                <Switch
                  id="adultFilter"
                  label="Adult"
                  switchToggle={[Function]}
                />
                <Switch
                  id="enLangFilter"
                  label="Original Language"
                  switchToggle={[Function]}
                />
              </div>,
              <div
                className="mt-1"
              >
                <Button
                  aria={Object {}}
                  click={[Function]}
                  label="Apply"
                />
              </div>,
            ],
            "className": "border-top border-primary mt-1 py-3 px-1",
            "id": "movie-filter",
            "role": "toolbar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filter options",
                "className": "text-uppercase",
              },
              "ref": null,
              "rendered": "Filter options",
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <label
                      aria-describedby="rating-filter"
                      aria-label="Rating filter"
                      htmlFor="rating-filter"
                    >
                      Rating (≥)
                      <input
                        aria-label="Rating range"
                        className="custom-range"
                        id="rating-filter"
                        max="10"
                        min="0"
                        name="ratingFilter"
                        onChange={[Function]}
                        step="0.5"
                        type="range"
                        value={0}
                      />
                      <span
                        className="text-primary position-absolute font-weight-bold ml-2"
                      >
                        0
                      </span>
                    </label>
                  </div>,
                  <Switch
                    id="adultFilter"
                    label="Adult"
                    switchToggle={[Function]}
                  />,
                  <Switch
                    id="enLangFilter"
                    label="Original Language"
                    switchToggle={[Function]}
                  />,
                ],
                "className": "d-flex flex-wrap justify-content-between align-items-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <label
                      aria-describedby="rating-filter"
                      aria-label="Rating filter"
                      htmlFor="rating-filter"
                    >
                      Rating (≥)
                      <input
                        aria-label="Rating range"
                        className="custom-range"
                        id="rating-filter"
                        max="10"
                        min="0"
                        name="ratingFilter"
                        onChange={[Function]}
                        step="0.5"
                        type="range"
                        value={0}
                      />
                      <span
                        className="text-primary position-absolute font-weight-bold ml-2"
                      >
                        0
                      </span>
                    </label>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-describedby": "rating-filter",
                      "aria-label": "Rating filter",
                      "children": Array [
                        "Rating (≥)",
                        <input
                          aria-label="Rating range"
                          className="custom-range"
                          id="rating-filter"
                          max="10"
                          min="0"
                          name="ratingFilter"
                          onChange={[Function]}
                          step="0.5"
                          type="range"
                          value={0}
                        />,
                        <span
                          className="text-primary position-absolute font-weight-bold ml-2"
                        >
                          0
                        </span>,
                      ],
                      "htmlFor": "rating-filter",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Rating (≥)",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "aria-label": "Rating range",
                          "className": "custom-range",
                          "id": "rating-filter",
                          "max": "10",
                          "min": "0",
                          "name": "ratingFilter",
                          "onChange": [Function],
                          "step": "0.5",
                          "type": "range",
                          "value": 0,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": 0,
                          "className": "text-primary position-absolute font-weight-bold ml-2",
                        },
                        "ref": null,
                        "rendered": 0,
                        "type": "span",
                      },
                    ],
                    "type": "label",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "adultFilter",
                    "label": "Adult",
                    "switchToggle": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "enLangFilter",
                    "label": "Original Language",
                    "switchToggle": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  aria={Object {}}
                  click={[Function]}
                  label="Apply"
                />,
                "className": "mt-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "aria": Object {},
                  "click": [Function],
                  "label": "Apply",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="d-flex justify-content-between align-items-center"
          >
            <h6
              className="mb-0"
            >
              Popular movies
            </h6>
            <Button
              aria={
                Object {
                  "aria-controls": "movie-filter",
                  "aria-expanded": false,
                }
              }
              click={[Function]}
              label="Filter"
            />
          </section>,
          <Collapse
            appear={false}
            dimension="height"
            getDimensionValue={[Function]}
            in={false}
            mountOnEnter={false}
            timeout={300}
            unmountOnExit={false}
          >
            <div
              aria-label="Filter options"
              className="border-top border-primary mt-1 py-3 px-1"
              id="movie-filter"
              role="toolbar"
            >
              <h6
                className="text-uppercase"
              >
                Filter options
              </h6>
              <div
                className="d-flex flex-wrap justify-content-between align-items-center"
              >
                <div>
                  <label
                    aria-describedby="rating-filter"
                    aria-label="Rating filter"
                    htmlFor="rating-filter"
                  >
                    Rating (≥)
                    <input
                      aria-label="Rating range"
                      className="custom-range"
                      id="rating-filter"
                      max="10"
                      min="0"
                      name="ratingFilter"
                      onChange={[Function]}
                      step="0.5"
                      type="range"
                      value={0}
                    />
                    <span
                      className="text-primary position-absolute font-weight-bold ml-2"
                    >
                      0
                    </span>
                  </label>
                </div>
                <Switch
                  id="adultFilter"
                  label="Adult"
                  switchToggle={[Function]}
                />
                <Switch
                  id="enLangFilter"
                  label="Original Language"
                  switchToggle={[Function]}
                />
              </div>
              <div
                className="mt-1"
              >
                <Button
                  aria={Object {}}
                  click={[Function]}
                  label="Apply"
                />
              </div>
            </div>
          </Collapse>,
        ],
        "className": "bg-secondary px-3 rounded",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h6
                className="mb-0"
              >
                Popular movies
              </h6>,
              <Button
                aria={
                  Object {
                    "aria-controls": "movie-filter",
                    "aria-expanded": false,
                  }
                }
                click={[Function]}
                label="Filter"
              />,
            ],
            "className": "d-flex justify-content-between align-items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Popular movies",
                "className": "mb-0",
              },
              "ref": null,
              "rendered": "Popular movies",
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "aria": Object {
                  "aria-controls": "movie-filter",
                  "aria-expanded": false,
                },
                "click": [Function],
                "label": "Filter",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <div
              aria-label="Filter options"
              className="border-top border-primary mt-1 py-3 px-1"
              id="movie-filter"
              role="toolbar"
            >
              <h6
                className="text-uppercase"
              >
                Filter options
              </h6>
              <div
                className="d-flex flex-wrap justify-content-between align-items-center"
              >
                <div>
                  <label
                    aria-describedby="rating-filter"
                    aria-label="Rating filter"
                    htmlFor="rating-filter"
                  >
                    Rating (≥)
                    <input
                      aria-label="Rating range"
                      className="custom-range"
                      id="rating-filter"
                      max="10"
                      min="0"
                      name="ratingFilter"
                      onChange={[Function]}
                      step="0.5"
                      type="range"
                      value={0}
                    />
                    <span
                      className="text-primary position-absolute font-weight-bold ml-2"
                    >
                      0
                    </span>
                  </label>
                </div>
                <Switch
                  id="adultFilter"
                  label="Adult"
                  switchToggle={[Function]}
                />
                <Switch
                  id="enLangFilter"
                  label="Original Language"
                  switchToggle={[Function]}
                />
              </div>
              <div
                className="mt-1"
              >
                <Button
                  aria={Object {}}
                  click={[Function]}
                  label="Apply"
                />
              </div>
            </div>,
            "dimension": "height",
            "getDimensionValue": [Function],
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Filter options",
              "children": Array [
                <h6
                  className="text-uppercase"
                >
                  Filter options
                </h6>,
                <div
                  className="d-flex flex-wrap justify-content-between align-items-center"
                >
                  <div>
                    <label
                      aria-describedby="rating-filter"
                      aria-label="Rating filter"
                      htmlFor="rating-filter"
                    >
                      Rating (≥)
                      <input
                        aria-label="Rating range"
                        className="custom-range"
                        id="rating-filter"
                        max="10"
                        min="0"
                        name="ratingFilter"
                        onChange={[Function]}
                        step="0.5"
                        type="range"
                        value={0}
                      />
                      <span
                        className="text-primary position-absolute font-weight-bold ml-2"
                      >
                        0
                      </span>
                    </label>
                  </div>
                  <Switch
                    id="adultFilter"
                    label="Adult"
                    switchToggle={[Function]}
                  />
                  <Switch
                    id="enLangFilter"
                    label="Original Language"
                    switchToggle={[Function]}
                  />
                </div>,
                <div
                  className="mt-1"
                >
                  <Button
                    aria={Object {}}
                    click={[Function]}
                    label="Apply"
                  />
                </div>,
              ],
              "className": "border-top border-primary mt-1 py-3 px-1",
              "id": "movie-filter",
              "role": "toolbar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filter options",
                  "className": "text-uppercase",
                },
                "ref": null,
                "rendered": "Filter options",
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <label
                        aria-describedby="rating-filter"
                        aria-label="Rating filter"
                        htmlFor="rating-filter"
                      >
                        Rating (≥)
                        <input
                          aria-label="Rating range"
                          className="custom-range"
                          id="rating-filter"
                          max="10"
                          min="0"
                          name="ratingFilter"
                          onChange={[Function]}
                          step="0.5"
                          type="range"
                          value={0}
                        />
                        <span
                          className="text-primary position-absolute font-weight-bold ml-2"
                        >
                          0
                        </span>
                      </label>
                    </div>,
                    <Switch
                      id="adultFilter"
                      label="Adult"
                      switchToggle={[Function]}
                    />,
                    <Switch
                      id="enLangFilter"
                      label="Original Language"
                      switchToggle={[Function]}
                    />,
                  ],
                  "className": "d-flex flex-wrap justify-content-between align-items-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <label
                        aria-describedby="rating-filter"
                        aria-label="Rating filter"
                        htmlFor="rating-filter"
                      >
                        Rating (≥)
                        <input
                          aria-label="Rating range"
                          className="custom-range"
                          id="rating-filter"
                          max="10"
                          min="0"
                          name="ratingFilter"
                          onChange={[Function]}
                          step="0.5"
                          type="range"
                          value={0}
                        />
                        <span
                          className="text-primary position-absolute font-weight-bold ml-2"
                        >
                          0
                        </span>
                      </label>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-describedby": "rating-filter",
                        "aria-label": "Rating filter",
                        "children": Array [
                          "Rating (≥)",
                          <input
                            aria-label="Rating range"
                            className="custom-range"
                            id="rating-filter"
                            max="10"
                            min="0"
                            name="ratingFilter"
                            onChange={[Function]}
                            step="0.5"
                            type="range"
                            value={0}
                          />,
                          <span
                            className="text-primary position-absolute font-weight-bold ml-2"
                          >
                            0
                          </span>,
                        ],
                        "htmlFor": "rating-filter",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Rating (≥)",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "aria-label": "Rating range",
                            "className": "custom-range",
                            "id": "rating-filter",
                            "max": "10",
                            "min": "0",
                            "name": "ratingFilter",
                            "onChange": [Function],
                            "step": "0.5",
                            "type": "range",
                            "value": 0,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": 0,
                            "className": "text-primary position-absolute font-weight-bold ml-2",
                          },
                          "ref": null,
                          "rendered": 0,
                          "type": "span",
                        },
                      ],
                      "type": "label",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "id": "adultFilter",
                      "label": "Adult",
                      "switchToggle": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "id": "enLangFilter",
                      "label": "Original Language",
                      "switchToggle": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    aria={Object {}}
                    click={[Function]}
                    label="Apply"
                  />,
                  "className": "mt-1",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "aria": Object {},
                    "click": [Function],
                    "label": "Apply",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
